src/libstore/content-address.cc

```diff
@@ -11,6 +11,9 @@ std::string makeFileIngestionPrefix(FileIngestionMethod m)
         return "";
     case FileIngestionMethod::Recursive:
         return "r:";
+    case FileIngestionMethod::Git:
```

Comment:
Isn't this still "recursive", but the "content address prefix" is `git`?

I peeked at the Nix codebase, and the CA field in a `.narinfo`, and:

```
    // Semantically, they can be split into the following components:
    //  - "content address prefix". Currently, "fixed" and "text" are supported.
    //  - "hash mode". Currently, "flat" and "recursive" are supported.
    //  - "hash type". The underlying hash function used.
    //    Currently, sha1, md5, sha256, sha512.
    //  - "digest". The digest itself.
```

I think it'd be cleaner if we'd have a "git" content address prefix, and "recursive" / "flat" distinguish if we refer to a tree or blob object (see the other comments).

Reply:
Thinking about this a bit more, I'm somewhat convinced "recursive" is a bad name, and probably should simply have been called "nar" hashing.

The scope of this should be reduced to only do git tree objects, and accept some store paths can't be represented with it (like already the case for symlinks and executable files).

Reply:
Yeah it definitely should be called Nar.